### PR TITLE
target/ppc: Implement vsldbi/vsrdbi instructions

### PR DESCRIPTION
**Pull request by mferst**

### CODE REVIEW

File: target/ppc/translate/vmx-impl.c.inc [hunk 1]
```diff
@@ -1257,6 +1257,80 @@ static void gen_vsldoi(DisasContext *ctx)
...
+static bool trans_VSLDBI(DisasContext *ctx, arg_VN *a)
+{
+    TCGv_i64 t0, t1, t2;
+
+    REQUIRE_INSNS_FLAGS2(ctx, ISA310);
+    REQUIRE_VECTOR(ctx);
+
+    t0 = tcg_temp_new_i64();
+    t1 = tcg_temp_new_i64();
+    t2 = tcg_temp_new_i64();
```
Comment (lffpires) [inline target/ppc/translate/vmx-impl.c.inc:1269]: Maybe move ``t2 = tcg_temp_new_i64()`` and ``tcg_temp_free_i64(t2)`` inside the ``if (a->sh != 0)`` block?

File: target/ppc/translate/vmx-impl.c.inc [hunk 2]
```diff
@@ -1257,6 +1257,80 @@ static void gen_vsldoi(DisasContext *ctx)
...
+static bool trans_VSRDBI(DisasContext *ctx, arg_VN *a)
+{
+    TCGv_i64 t2, t1, t0;
+
+    REQUIRE_INSNS_FLAGS2(ctx, ISA310);
+    REQUIRE_VECTOR(ctx);
+
+    t0 = tcg_temp_new_i64();
+    t1 = tcg_temp_new_i64();
+    t2 = tcg_temp_new_i64();
```
Comment (lffpires) [inline target/ppc/translate/vmx-impl.c.inc:1306]: Same here

**Review by lffpires — APPROVED** [review]
LGTM! Minor suggestions.

File: target/ppc/translate/vmx-impl.c.inc (hunk 1 above)
Comment (mferst) [reply target/ppc/translate/vmx-impl.c.inc]: Fixed in https://github.com/PPC64/qemu/pull/18/commits/69a1f824a0db9d22f122f26e2ed86da0b34e52ad

File: target/ppc/translate/vmx-impl.c.inc (hunk 2 above)
Comment (mferst) [reply target/ppc/translate/vmx-impl.c.inc]: Fixed in https://github.com/PPC64/qemu/pull/18/commits/69a1f824a0db9d22f122f26e2ed86da0b34e52ad

**Review by alqotel — APPROVED** [review]

**Comment by mferst** [comment]
Merged in upstream.

**Pull request closed without merging**